### ISSUE TITLE
Dockerfile build fails

### ISSUE THREAD

**Issue by sonamoo** [openstf/stf#923]
I am using the latest version of STF. I cloned this repository and ran docker build command and I got error messages.

```Sending build context to Docker daemon  9.289MB
Step 1/11 : FROM ubuntu:16.04
 ---> 7aa3602ab41e
Step 2/11 : ENV PATH /app/bin:$PATH
 ---> Using cache
 ---> 6547db7d767f
Step 3/11 : WORKDIR /app
 ---> Using cache
 ---> 3d67caa6f63a
Step 4/11 : EXPOSE 3000
 ---> Using cache
 ---> 894ecca7c168
Step 5/11 : RUN export DEBIAN_FRONTEND=noninteractive &&     useradd --system       --create-home       --shell /usr/sbin/nologin       stf-build &&     useradd --system       --create-home       --shell /usr/sbin/nologin       stf &&     sed -i'' 's@http://archive.ubuntu.com/ubuntu/@mirror://mirrors.ubuntu.com/mirrors.txt@' /etc/apt/sources.list &&     apt-get update &&     apt-get -y install wget python build-essential &&     cd /tmp &&     wget --progress=dot:mega       https://nodejs.org/dist/v6.11.2/node-v6.11.2-linux-x64.tar.xz &&     tar -xJf node-v*.tar.xz --strip-components 1 -C /usr/local &&     rm node-v*.tar.xz &&     su stf-build -s /bin/bash -c '/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js install' &&     apt-get -y install libzmq3-dev libprotobuf-dev git graphicsmagick yasm &&     apt-get clean &&     rm -rf /var/cache/apt/* /var/lib/apt/lists/*
 ---> Running in 13d66f65b44e
Err:1 http://security.ubuntu.com/ubuntu xenial-security InRelease
  Temporary failure resolving 'security.ubuntu.com'
Ign:2 mirror://mirrors.ubuntu.com/mirrors.txt xenial InRelease
Ign:3 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates InRelease
Ign:4 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports InRelease
Ign:5 mirror://mirrors.ubuntu.com/mirrors.txt xenial Release
Ign:6 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates Release
Ign:7 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports Release
Ign:8 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe Sources
Ign:9 mirror://mirrors.ubuntu.com/mirrors.txt xenial/main amd64 Packages
Ign:10 mirror://mirrors.ubuntu.com/mirrors.txt xenial/main all Packages
Ign:11 mirror://mirrors.ubuntu.com/mirrors.txt xenial/restricted amd64 Packages
Ign:12 mirror://mirrors.ubuntu.com/mirrors.txt xenial/restricted all Packages
Ign:13 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe amd64 Packages
Ign:14 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe all Packages
Ign:15 mirror://mirrors.ubuntu.com/mirrors.txt xenial/multiverse amd64 Packages
Ign:16 mirror://mirrors.ubuntu.com/mirrors.txt xenial/multiverse all Packages
Ign:17 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe Sources
Ign:18 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/main amd64 Packages
Ign:19 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/main all Packages
Ign:20 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/restricted amd64 Packages
Ign:21 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/restricted all Packages
Ign:22 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe amd64 Packages
Ign:23 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe all Packages
Ign:24 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/multiverse amd64 Packages
Ign:25 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/multiverse all Packages
Ign:26 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/main amd64 Packages
Ign:27 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/main all Packages
Ign:28 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/restricted amd64 Packages
Ign:29 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/restricted all Packages
Ign:30 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/universe amd64 Packages
Ign:31 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/universe all Packages
Ign:32 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/multiverse amd64 Packages
Ign:33 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/multiverse all Packages
Ign:8 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe Sources
Ign:9 mirror://mirrors.ubuntu.com/mirrors.txt xenial/main amd64 Packages
Ign:10 mirror://mirrors.ubuntu.com/mirrors.txt xenial/main all Packages
Ign:11 mirror://mirrors.ubuntu.com/mirrors.txt xenial/restricted amd64 Packages
Ign:12 mirror://mirrors.ubuntu.com/mirrors.txt xenial/restricted all Packages
Ign:13 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe amd64 Packages
Ign:14 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe all Packages
Ign:15 mirror://mirrors.ubuntu.com/mirrors.txt xenial/multiverse amd64 Packages
Ign:16 mirror://mirrors.ubuntu.com/mirrors.txt xenial/multiverse all Packages
Ign:17 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe Sources
Ign:18 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/main amd64 Packages
Ign:19 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/main all Packages
Ign:20 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/restricted amd64 Packages
Ign:21 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/restricted all Packages
Ign:22 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe amd64 Packages
Ign:23 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe all Packages
Ign:24 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/multiverse amd64 Packages
Ign:25 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/multiverse all Packages
Ign:26 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/main amd64 Packages
Ign:27 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/main all Packages
Ign:28 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/restricted amd64 Packages
Ign:29 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/restricted all Packages
Ign:30 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/universe amd64 Packages
Ign:31 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/universe all Packages
Ign:32 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/multiverse amd64 Packages
Ign:33 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/multiverse all Packages
Ign:8 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe Sources
Ign:9 mirror://mirrors.ubuntu.com/mirrors.txt xenial/main amd64 Packages
Ign:10 mirror://mirrors.ubuntu.com/mirrors.txt xenial/main all Packages
Ign:11 mirror://mirrors.ubuntu.com/mirrors.txt xenial/restricted amd64 Packages
Ign:12 mirror://mirrors.ubuntu.com/mirrors.txt xenial/restricted all Packages
Ign:13 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe amd64 Packages
Ign:14 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe all Packages
Ign:15 mirror://mirrors.ubuntu.com/mirrors.txt xenial/multiverse amd64 Packages
Ign:16 mirror://mirrors.ubuntu.com/mirrors.txt xenial/multiverse all Packages
Ign:17 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe Sources
Ign:18 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/main amd64 Packages
Ign:19 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/main all Packages
Ign:20 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/restricted amd64 Packages
Ign:21 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/restricted all Packages
Ign:22 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe amd64 Packages
Ign:23 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe all Packages
Ign:24 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/multiverse amd64 Packages
Ign:25 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/multiverse all Packages
Ign:26 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/main amd64 Packages
Ign:27 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/main all Packages
Ign:28 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/restricted amd64 Packages
Ign:29 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/restricted all Packages
Ign:30 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/universe amd64 Packages
Ign:31 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/universe all Packages
Ign:32 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/multiverse amd64 Packages
Ign:33 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/multiverse all Packages
Ign:8 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe Sources
Ign:9 mirror://mirrors.ubuntu.com/mirrors.txt xenial/main amd64 Packages
Ign:10 mirror://mirrors.ubuntu.com/mirrors.txt xenial/main all Packages
Ign:11 mirror://mirrors.ubuntu.com/mirrors.txt xenial/restricted amd64 Packages
Ign:12 mirror://mirrors.ubuntu.com/mirrors.txt xenial/restricted all Packages
Ign:13 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe amd64 Packages
Ign:14 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe all Packages
Ign:15 mirror://mirrors.ubuntu.com/mirrors.txt xenial/multiverse amd64 Packages
Ign:16 mirror://mirrors.ubuntu.com/mirrors.txt xenial/multiverse all Packages
Ign:17 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe Sources
Ign:18 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/main amd64 Packages
Ign:19 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/main all Packages
Ign:20 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/restricted amd64 Packages
Ign:21 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/restricted all Packages
Ign:22 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe amd64 Packages
Ign:23 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe all Packages
Ign:24 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/multiverse amd64 Packages
Ign:25 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/multiverse all Packages
Ign:26 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/main amd64 Packages
Ign:27 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/main all Packages
Ign:28 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/restricted amd64 Packages
Ign:29 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/restricted all Packages
Ign:30 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/universe amd64 Packages
Ign:31 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/universe all Packages
Ign:32 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/multiverse amd64 Packages
Ign:33 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/multiverse all Packages
Ign:8 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe Sources
Ign:9 mirror://mirrors.ubuntu.com/mirrors.txt xenial/main amd64 Packages
Ign:10 mirror://mirrors.ubuntu.com/mirrors.txt xenial/main all Packages
Ign:11 mirror://mirrors.ubuntu.com/mirrors.txt xenial/restricted amd64 Packages
Ign:12 mirror://mirrors.ubuntu.com/mirrors.txt xenial/restricted all Packages
Ign:13 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe amd64 Packages
Ign:14 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe all Packages
Ign:15 mirror://mirrors.ubuntu.com/mirrors.txt xenial/multiverse amd64 Packages
Ign:16 mirror://mirrors.ubuntu.com/mirrors.txt xenial/multiverse all Packages
Ign:17 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe Sources
Ign:18 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/main amd64 Packages
Ign:19 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/main all Packages
Ign:20 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/restricted amd64 Packages
Ign:21 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/restricted all Packages
Ign:22 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe amd64 Packages
Ign:23 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe all Packages
Ign:24 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/multiverse amd64 Packages
Ign:25 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/multiverse all Packages
Ign:26 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/main amd64 Packages
Ign:27 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/main all Packages
Ign:28 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/restricted amd64 Packages
Ign:29 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/restricted all Packages
Ign:30 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/universe amd64 Packages
Ign:31 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/universe all Packages
Ign:32 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/multiverse amd64 Packages
Ign:33 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/multiverse all Packages
Err:8 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe Sources
  No mirror file '/var/lib/apt/mirrors/mirrors.ubuntu.com_mirrors.txt' found  [Mirror: ]
Ign:9 mirror://mirrors.ubuntu.com/mirrors.txt xenial/main amd64 Packages
Ign:10 mirror://mirrors.ubuntu.com/mirrors.txt xenial/main all Packages
Ign:11 mirror://mirrors.ubuntu.com/mirrors.txt xenial/restricted amd64 Packages
Ign:12 mirror://mirrors.ubuntu.com/mirrors.txt xenial/restricted all Packages
Ign:13 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe amd64 Packages
Ign:14 mirror://mirrors.ubuntu.com/mirrors.txt xenial/universe all Packages
Ign:15 mirror://mirrors.ubuntu.com/mirrors.txt xenial/multiverse amd64 Packages
Ign:16 mirror://mirrors.ubuntu.com/mirrors.txt xenial/multiverse all Packages
Err:17 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe Sources
  No mirror file '/var/lib/apt/mirrors/mirrors.ubuntu.com_mirrors.txt' found  [Mirror: ]
Ign:18 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/main amd64 Packages
Ign:19 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/main all Packages
Ign:20 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/restricted amd64 Packages
Ign:21 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/restricted all Packages
Ign:22 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe amd64 Packages
Ign:23 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/universe all Packages
Ign:24 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/multiverse amd64 Packages
Ign:25 mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates/multiverse all Packages
Err:26 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/main amd64 Packages
  No mirror file '/var/lib/apt/mirrors/mirrors.ubuntu.com_mirrors.txt' found  [Mirror: ]
Ign:27 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/main all Packages
Ign:28 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/restricted amd64 Packages
Ign:29 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/restricted all Packages
Ign:30 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/universe amd64 Packages
Ign:31 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/universe all Packages
Ign:32 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/multiverse amd64 Packages
Ign:33 mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports/multiverse all Packages
Reading package lists...
W: The repository 'mirror://mirrors.ubuntu.com/mirrors.txt xenial Release' does not have a Release file.
W: The repository 'mirror://mirrors.ubuntu.com/mirrors.txt xenial-updates Release' does not have a Release file.
W: The repository 'mirror://mirrors.ubuntu.com/mirrors.txt xenial-backports Release' does not have a Release file.
W: Failed to fetch http://security.ubuntu.com/ubuntu/dists/xenial-security/InRelease  Temporary failure resolving 'security.ubuntu.com'
E: Failed to fetch mirror://mirrors.ubuntu.com/mirrors.txt/dists/xenial/universe/source/Sources  No mirror file '/var/lib/apt/mirrors/mirrors.ubuntu.com_mirrors.txt' found  [Mirror: ]
E: Failed to fetch mirror://mirrors.ubuntu.com/mirrors.txt/dists/xenial-updates/universe/source/Sources  No mirror file '/var/lib/apt/mirrors/mirrors.ubuntu.com_mirrors.txt' found  [Mirror: ]
E: Failed to fetch mirror://mirrors.ubuntu.com/mirrors.txt/dists/xenial-backports/main/binary-amd64/Packages  No mirror file '/var/lib/apt/mirrors/mirrors.ubuntu.com_mirrors.txt' found  [Mirror: ]
W: Some index files failed to download. They have been ignored, or old ones used instead.
The command '/bin/sh -c export DEBIAN_FRONTEND=noninteractive &&     useradd --system       --create-home       --shell /usr/sbin/nologin       stf-build &&     useradd --system       --create-home       --shell /usr/sbin/nologin       stf &&     sed -i'' 's@http://archive.ubuntu.com/ubuntu/@mirror://mirrors.ubuntu.com/mirrors.txt@' /etc/apt/sources.list &&     apt-get update &&     apt-get -y install wget python build-essential &&     cd /tmp &&     wget --progress=dot:mega       https://nodejs.org/dist/v6.11.2/node-v6.11.2-linux-x64.tar.xz &&     tar -xJf node-v*.tar.xz --strip-components 1 -C /usr/local &&     rm node-v*.tar.xz &&     su stf-build -s /bin/bash -c '/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js install' &&     apt-get -y install libzmq3-dev libprotobuf-dev git graphicsmagick yasm &&     apt-get clean &&     rm -rf /var/cache/apt/* /var/lib/apt/lists/*' returned a non-zero code: 100
```

**Am I doing something wrong? Is there something wrong in Regex in the command?**

`sed -i'' 's@http://archive.ubuntu.com/ubuntu/@mirror://mirrors.ubuntu.com/mirrors.txt@' /etc/apt/sources.list`

Thank you so much!

**Comment by koral--** [comment]
```Err:1 http://security.ubuntu.com/ubuntu xenial-security InRelease
  Temporary failure resolving 'security.ubuntu.com'
```
It looks like network connectivity or DNS issue. Check DNS settings and whether you are able to reach security.ubuntu.com from host you are building on.

**Comment by sonamoo** [comment]
The Ubuntu image I have did not have curl or any text editor installed. However I could check that this ubuntu container could not access the internet. I ran `apt-get update` and this command failed with this error message, `Temporary failure resolving 'archive.ubuntu.com'` To find out the causes, I ran `cat /etc/resolve.conf` and the result was 
```
search my.companyName.com

nameserver 8.8.8.8
nameserver 8.8.4.4
```

Is this happening because of this line `search my.companyName.com`?

Will getting rid of this line solve the issue? This ubuntu image does not have anything and can not even do 'apt-get' so maybe I should build a custom Ubuntu image and put text editor and then edit the line?

**Comment by gounthar** [comment]
Are you using a corporate proxy of some sort?

**Comment by sonamoo** [comment]
@gounthar 
Yes, I am using STF at my company and using the provided wifi network.

**Comment by gounthar** [comment]
If you know your proxy machine, you could configure your docker to use it.

### Docker

If you try to download a Docker image with `docker pull ...` it will fail because Docker doesn't use the proxy set for the user.

So you will have to create a few directories and files to let Docker know it should use your proxy:

```bash
$ sudo mkdir -p /etc/systemd/system/docker.service.d/
$ sudo vi /etc/systemd/system/docker.service.d/http-proxy.conf
$ sudo vi /etc/systemd/system/docker.service.d/https-proxy.conf
```

`/etc/systemd/system/docker.service.d/https-proxy.conf` will contain:

```bash
[Service]
Environment="HTTPS_PROXY=http://your_proxy:your_proxy_port" "NO_PROXY=localhost,127.0.0.1,*.your.local.net"
```

The `http` version of this file will have the same proxy settings (just remove the `S` of `HTTPS`). 
Once you're done with the modifications, you will have to tell Docker to restart to take this into account:

```bash
$ sudo systemctl daemon-reload
$ sudo systemctl restart docker
```

Then you'll be able to check that your modifications are correct:

```bash
$ systemctl show --property=Environment docker
Environment=HTTP_PROXY=http://your_proxy:your_proxy_port/ NO_PROXY=localhost,127.0.0.1,your.local.net HTTPS_PROXY=http://your_proxy:your_proxy_port/
```

**Comment by sonamoo** [comment]
Hi @gounthar,

Thank you for suggesting! I think I was not sure about what causes the issues. I solved this problem by adding my local DNS server address to the Docker containers. 

I found out that the containers could ping 8.8.8.8 but not www.google.com so start digging the dns options for docker daemon.

Many people suggested to fix /etc/default/docker file to add more options to the docker, and I tried it many times but failed. I realized that the file is for the OS that uses upstart and `sysvinit`, but not for the OS that runs `Systemd`. As the docker team suggested && and you suggested I created `/etc/systemd/system/docker.service.d` directory and then created `conf `file to add a `flag `when `systemd `initializes Docker daemon. I could simply add` --dns <DNS address>` and then restarted the docker.

And it worked! 


Thank you again @koral-- , @gounthar and I apologize for opening up issues with non STF stuff :)
